infrastructure/src/main/java/org/corfudb/infrastructure/log/SegmentHandle.java

```diff
@@ -23,23 +23,22 @@
 @Slf4j
 @Data
 class SegmentHandle {
-    final long segment;
+    private final long segment;
 
     @NonNull
-    final FileChannel writeChannel;
+    private final FileChannel writeChannel;
 
     @NonNull
-    final FileChannel readChannel;
+    private final FileChannel readChannel;
 
     @NonNull
-    String fileName;
+    private final Path fileName;
 
     private final Map<Long, AddressMetaData> knownAddresses = new ConcurrentHashMap<>();
     private final Set<Long> trimmedAddresses = Collections.newSetFromMap(new ConcurrentHashMap<>());
     private final Set<Long> pendingTrims = Collections.newSetFromMap(new ConcurrentHashMap<>());
     private volatile int refCount = 0;
```

Comment:
You can remove trimmedAddresses and pendingTrims they were used for sparse trimming. But that feature has been removed.

Reply:
let's create another PR for that